runtime/devnet/src/config/api/versioning.rs

```diff
@@ -0,0 +1,220 @@
+use sp_runtime::ModuleError;
+
+use super::*;
+
+type Version = u8;
+
+/// Versioned runtime calls.
+#[derive(Decode, Debug)]
+pub enum VersionedRuntimeCall {
+	/// Version zero of runtime calls.
+	#[codec(index = 0)]
+	V0(RuntimeCall),
+}
+
+impl From<VersionedRuntimeCall> for RuntimeCall {
+	fn from(value: VersionedRuntimeCall) -> Self {
+		// Allows mapping from some previous runtime call shape to a current valid runtime call
+		match value {
+			VersionedRuntimeCall::V0(call) => call,
+		}
+	}
+}
+
+/// Versioned runtime state reads.
+#[derive(Decode, Debug)]
+pub enum VersionedRuntimeRead {
+	/// Version zero of runtime state reads.
+	#[codec(index = 0)]
+	V0(RuntimeRead),
+}
+
+impl From<VersionedRuntimeRead> for RuntimeRead {
+	fn from(value: VersionedRuntimeRead) -> Self {
+		// Allows mapping from some previous runtime read shape to a current valid runtime read
+		match value {
+			VersionedRuntimeRead::V0(read) => read,
+		}
+	}
+}
+
+/// Versioned runtime state read results.
+#[derive(Debug)]
+pub enum VersionedRuntimeResult {
+	/// Version zero of runtime read results.
+	V0(RuntimeResult),
+}
+
+impl From<(RuntimeResult, Version)> for VersionedRuntimeResult {
+	fn from(value: (RuntimeResult, Version)) -> Self {
+		let (result, version) = value;
+		match version {
+			// Allows mapping from current `RuntimeResult` to a specific/prior version
+			0 => VersionedRuntimeResult::V0(result),
+			// TODO: should never occur due to version processing/validation when request received
+			_ => unimplemented!(),
```

Comment:
Shouldnt panic in runtime. Happen to address separately.

Reply:
#275